src/eig.rs

```diff
@@ -0,0 +1,50 @@
+//! Eigenvalue decomposition for Hermite matrices
+
+use ndarray::*;
+use crate::error::*;
+use crate::layout::*;
+use crate::types::*;
+
+/// Eigenvalue decomposition of general matrix reference
+pub trait Eig {
+    type EigVal;
+    type EigVec;
+    fn eig(&self) -> Result<(Self::EigVal, Self::EigVec)>;
```

Comment:
Please add a comment that this function calculate the **right** eigenvalue

Reply:
Ok, thanks.